### PR TITLE
Implement Jetpack Wizard banner

### PR DESCRIPTION
**Pull request by pidgeonman**
#### Changes proposed in this Pull Request:
This PR implements the Jetpack Wizard banner, which is basically the page 1 of the Wizard in a banner form to be displayed in all pages to encourage users to complete the Jetpack setup after a connection.

Screen Shot 2020-05-05 at 18 40 38

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
This is a new feature.

#### Testing instructions:
* Check out the branch, build,
* Connect Jetpack, but don't set it up yet.
* Add the following filter into wp_config.php:
```
add_filter( 'jetpack_show_setup_wizard', '__return_true' );
```
* Refresh your wp-admin, you should see the banner.

#### Proposed changelog entry for your changes:
* Implemented Jetpack Wizard banner.

### CODE REVIEW

**Review by robertf4 — COMMENTED** [review]
Some of the responsive behavior needs to be refined. 

For example at 700px wide the image overlaps the answer box and the question is no longer centered over the answers:

image

File: _inc/lib/class-jetpack-wizard.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	 *
+	 * @return bool
+	 */
+	public static function can_be_displayed() {
+		return apply_filters( 'jetpack_show_setup_wizard', false ) &&
+			Jetpack::is_active() &&
+			! self::is_finished();
+	}
+
+	// TODO: move save and get logic from the endpoint (update_setup_questionnaire() and get_setup_questionnaire()) to this class.
```
Comment (robertf4) [inline _inc/lib/class-jetpack-wizard.php:44]: You should either resolve the TODOs in this PR, or remove them and make a new task for them in Asana to do in another PR before merging.

**Review by robertf4 — COMMENTED** [review]
Tests well! Just a few minor comments.

File: _inc/lib/class-jetpack-wizard.php (hunk 1 above)
Comment (pidgeonman) [reply _inc/lib/class-jetpack-wizard.php]: Removed.

**Comment by pidgeonman** [comment]
> Some of the responsive behavior needs to be refined.
> 
> For example at 700px wide the image overlaps the answer box and the question is no longer centered over the answers:
> 
> image

Fixed!

**Comment by jeffgolenski** [comment]
@pidgeonman This is really starting to take shape. Thanks my friend. 

I've just a few small design tweaks for us to clean up and I think we'll be good here. Attaching!

jetpack-setup-wizard-pr

^ I also have this imagery live in figma, ping me for the link if you need it.

Cc @robertf4

File: scss/jetpack-wizard-banner.scss [hunk 2]
```diff
@@ -11,7 +11,7 @@
 	margin: 40px 20px 0 2px;
 	@include breakpoint( '<660px' ) {
 		margin: 45px 20px 0 2px;
-		padding-bottom: 50px;
+		padding-bottom: 0px;
```
Comment (jeffgolenski) [inline scss/jetpack-wizard-banner.scss:14]: "px" is unnecessary with 0

File: scss/jetpack-wizard-banner.scss [hunk 3]
```diff
@@ -0,0 +1,110 @@
+@import '../_inc/client/scss/mixin_breakpoint';
+@import '../_inc/client/scss/calypso-colors';
+
+
+.jp-wizard-banner {
+	text-align: left;
+	background: $white;
+	padding: 2rem;
+	box-shadow: 0px 2px 6px rgba( 0, 0, 0, 0.03 ), 0px 1px 2px rgba( 0, 0, 0, 0.03 );
+	border: 1px solid #dcdcde;
```
Comment (jeherve) [inline scss/jetpack-wizard-banner.scss:10]: I wonder if we should move away from Calypso colors and use Color studio instead; that will give us access to more up to date colors and more variables:

```suggestion
@import 'node_modules/@automattic/color-studio/dist/color-variables.scss';
@import '../_inc/client/scss/mixin_breakpoint';


.jp-wizard-banner {
	text-align: left;
	background: $studio-white;
	padding: 2rem;
	box-shadow: 0px 2px 6px rgba( 0, 0, 0, 0.03 ), 0px 1px 2px rgba( 0, 0, 0, 0.03 );
	border: 1px solid $studio-gray-5;
```

File: scss/jetpack-wizard-banner.scss [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	font-weight: 500;
+	font-size: 1.5em;
+	line-height: 150%;
+	@include breakpoint( '<660px' ) {
+		text-align: left;
+		font-weight: 400;
+	}
+}
+
+img {
```
Comment (jeherve) [inline scss/jetpack-wizard-banner.scss:29]: Maybe we should wrap all our CSS so it's more specific? Targetting just `img` here is a bit dangerous since we'll be enqueuing that CSS on almost all dashboard pages.

File: class-jetpack-wizard-banner.php [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+				width="200px"
+				height="200px"
+				src="
+				<?php
+				esc_attr_e( plugins_url( 'images/jetpack-powering-up.svg', JETPACK__PLUGIN_FILE ), 'jetpack' ); // phpcs:ignore WordPress.WP.I18n.NonSingularStringLiteralText
+				?>
+				"
+				alt="<?php esc_attr_e( 'A jetpack site powering up', 'jetpack' ); ?>"
+			/>
+			<h1 class="jp-wizard-banner-wizard-header">
```
Comment (jeherve) [inline class-jetpack-wizard-banner.php:139]: I don't know if this should be an `h1` heading. Admin page titles are already `h1`; maybe this should be `h2`?

File: class-jetpack-wizard-banner.php [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	/**
+	 * Renders the Wizard Banner
+	 */
+	public function render_banner() {
+		$jetpack_logo = new Jetpack_Logo();
+
+		?>
+		<div id="jp-wizard-banner" class="jp-wizard-banner">
```
Comment (jeherve) [inline class-jetpack-wizard-banner.php:119]: Given that this is replicating the contents of `_inc/client/setup-wizard/intro-page/index.jsx`, I think we should have a warning for now, in both files, that if you update one you must update the other as well.

On the long run, I think it would be nice if that content were retrieved from a single source to avoid duplicating things.

**Review by jeherve — CHANGES_REQUESTED** [review]
I gave it a try and would have a few remarks.

File: class-jetpack-wizard-banner.php [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+	 * Initialize hooks to display the banner
+	 */
+	public function maybe_initialize_hooks() {
+		// We already display the wizard at the Jetpack area.
+		if ( strpos( get_current_screen()->id, 'jetpack' ) !== false ) {
+			return;
+		}
+
+		// Kill if banner has been dismissed.
+		if ( Jetpack_Options::get_option( 'dismissed_wizard_banner' ) ) {
```
[the remark this answers is not in the document]
Comment (pidgeonman) [reply class-jetpack-wizard-banner.php:52]: This banner should be enabled by the jetpack_show_setup_wizard filter, which is already checked in https://github.com/Automattic/jetpack/pull/15690/files/101a01ac32b8cd061bcf6d08f8edcaaf45e2bea2#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R3738
Maybe I should move the `if ( strpos( get_current_screen()->id, 'jetpack' ) !== false ) {` check to that file.

[the remark this answers is not in the document]
Comment (pidgeonman) [reply class-jetpack-wizard-banner.php:52]: Refactored that section.

**Comment by jeffgolenski** [comment]
hey @pidgeonman I think we're almost there. Everything looks great, except I'm still experiencing the clear issue with the illustration in Chrome. Here's an animated gif to show you what I'm seeing: https://cloudup.com/cik8uSWXcqJ

File: _inc/lib/class-jetpack-wizard.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	/**
+	 * Can the Wizard be displayed?
+	 *
+	 * @return bool
+	 */
+	public static function can_be_displayed() {
+		/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
+		/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
```
Comment (jeherve) [inline _inc/lib/class-jetpack-wizard.php:40]: Something went wrong with GitHub's suggestion, it seems.

```suggestion
		/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
```

File: class-jetpack-wizard-banner.php [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+			<p class="jp-wizard-banner-wizard-paragraph">
+				<?php esc_html_e( 'Jetpack is a cloud-powered tool built by Automattic.', 'jetpack' ); ?>
+			</p>
+			<p class="jp-wizard-banner-wizard-paragraph">
+				<?php esc_html_e( 'Answer a few questions and we’ll help you secure, speed up, customize, and grow your WordPress website.', 'jetpack' ); ?>
+			</p>
+			<div class="jp-wizard-banner-wizard-intro-question">
+				<h2>
+					<?php
+					/* translators: %s is the site name */
```
Comment (jeherve) [inline class-jetpack-wizard-banner.php:175]:
```suggestion
```

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks good to me. I only have minor requests below. Then the only thing left should be Jeff's remark above.

File: scss/jetpack-wizard-banner.scss [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+		border-radius: 4px;
+	}
+
+	max-width: 720px;
+	padding: 1.5rem 0 1.5rem;
+	text-align: center;
+	margin-top: 2rem;
+
+	h2 {
+		margin-top: 0px;
```
Comment (jeffgolenski) [inline scss/jetpack-wizard-banner.scss:73]: Minor note: Anytime was have 0 we won't need the px

File: scss/jetpack-wizard-banner.scss [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+	@include breakpoint( '<480px' ) {
+		width: 100%;
+		margin: 0.5rem 0;
+	}
+}
+
+.jp-wizard-banner-wizard-skip-link {
+	text-decoration: underline !important;
+	color: #000000;
```
Comment (jeffgolenski) [inline scss/jetpack-wizard-banner.scss:107]: Can be reduced to #000. But we are importing the color variables, so we should use those instead of hard-coded hex colors

**Review by jeffgolenski — DISMISSED** [review]

**Review by jeherve — APPROVED** [review]
